cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -80,6 +83,23 @@ public SecurityAuthenticationHttpHandler(String realm, TokenValidator tokenValid
     this.discoveryServiceClient = discoveryServiceClient;
     this.discoverables = discoveryServiceClient.discover(Constants.Service.EXTERNAL_AUTHENTICATION);
     this.configuration = configuration;
+    this.bypassPattern = createMatcher(configuration.get(Constants.Security.Router.BYPASS_AUTHENTICATION_REGEX));
```

Comment:
I don't think anybody should be able to modify cdap-site.xml other than the CDAP user. security.enabled is in cConf... if anyone can modify it, she can simply turn off security. 

The more we discuss this the less I understand the division between cConf and sConf. The only sensitive one I can come up with is that cdap-security.xml contains configuration that is sensitive because it discloses security properties of other systems (such as passwords etc).